[NNC] Add documentation for splitWith APIs

navahgar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#58270 [NNC] Add documentation for splitWith APIs**
* #58269 [NNC] Make splitWithMask transform in-place
* #58268 [NNC] Make splitWithTail transform in-place

Differential Revision: [D28427226](https://our.internmc.facebook.com/intern/diff/D28427226)

navahgar committed:
[NNC] Add documentation for splitWith APIs

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28427226).

navahgar committed:
Update on "[NNC] Add documentation for splitWith APIs"


Differential Revision: [D28427226](https://our.internmc.facebook.com/intern/diff/D28427226)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28427226).

ZolotukhinM commented on torch/csrc/jit/tensorexpr/loopnest.h:
```diff
@@ -93,16 +93,65 @@ class TORCH_API LoopNest {
+  //
+  // The given loop will be transformed to the outer loop after splitting.
+  // So, the pointer to the input loop should be valid after splitting and
+  // will point to the outer loop.
```
Since the loop remain valid, we should remove the `outer` parameter from the API. It would be also good to mention the `inner` and `tail` arguments in this comment.

navahgar replied:
Removed `outer` parameter in both the APIs. Added documentation about `inner` and `tail` as well.

ZolotukhinM commented on torch/csrc/jit/tensorexpr/loopnest.h:
```diff
@@ -93,16 +93,65 @@ class TORCH_API LoopNest {
+  //
+  // The given loop will be transformed to the outer loop after splitting.
+  // So, the pointer to the input loop should be valid after splitting and
+  // will point to the outer loop.
```
Same here, we should remove `outer` from teh API since it's now the same as the input loop.

Also, we might consider adding an API to conveniently retrieve the only child loop and replace this API with that. E.g.:
```
splitWithMask(my_loop, 8)
inner_loop = my_loop.getSingleChildLoop() # will throw an error if there is more than 1 stmts in the body or if it's not a loop
```

navahgar replied:
Removed `outer` parameter.

Good idea about the single child loop API. Will do that separately.

navahgar committed:
Update on "[NNC] Add documentation for splitWith APIs"


Differential Revision: [D28427226](https://our.internmc.facebook.com/intern/diff/D28427226)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28427226).

ZolotukhinM reviewed (approved):
👍

navahgar committed:
Update on "[NNC] Add documentation for splitWith APIs"


Differential Revision: [D28427226](https://our.internmc.facebook.com/intern/diff/D28427226)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28427226).

navahgar committed:
Update on "[NNC] Add documentation for splitWith APIs"


Differential Revision: [D28427226](https://our.internmc.facebook.com/intern/diff/D28427226)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28427226).

navahgar committed:
Update on "[NNC] Add documentation for splitWith APIs"


Differential Revision: [D28427226](https://our.internmc.facebook.com/intern/diff/D28427226)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28427226).
